weDevsOfficial/dokan
# feat: one step product creation

### All Submissions:






### Changes proposed in this Pull Request:



* Added One step product creation-related settings
*  Based on Admin settings product status select field changed to **Online** or **Pending Review**
* One-page/single-step product addition added. this closes #1508


![Screenshot 2023-08-14 at 10 19 18 AM](https://github.com/weDevsOfficial/dokan/assets/5476784/cfb83287-e67e-4069-9cf6-e9ec89a3a6a3)

![image](https://github.com/weDevsOfficial/dokan/assets/5476784/7e50eb62-d62b-4883-b8b9-e5179ea69053)


### How to test the changes in this Pull Request:

1. Please check issue #1508
1. related https://github.com/weDevsOfficial/dokan-pro/pull/2434


### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

**feat**: One Step product addition
Previously, vendors had to use a product popup or a limited field page to create a product. Now Dokan is giving one step single-page product addition experience.


### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.


### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.
